src/messaging/ExchangeMgr.cpp

```diff
@@ -266,7 +265,8 @@ void ExchangeManager::OnMessageReceived(const PacketHeader & packetHeader, const
     // an ack to the peer.
     else if (!payloadHeader.NeedsAck())
     {
-        ExitNow(err = CHIP_ERROR_UNSOLICITED_MSG_NO_ORIGINATOR);
+        ChipLogError(ExchangeManager, "OnMessageReceived failed, err = %s", ErrorStr(CHIP_ERROR_UNSOLICITED_MSG_NO_ORIGINATOR));
```

Comment:
If we're doing this log in this specific spot, how about we log what's really happening?  Something like:
```suggestion
        ChipLogError(ExchangeManager, "Dropping unsolicited message that claims to be a response");
```

Reply:
I'm re-using the same error message to optimize code size, as the error string is typically 50-100 bytes, this is not a small impact.

Reply:
OK.  Worth a comment in the code then, so someone doesn't "improve" it.

Reply:
Added a comment for it